File: django/app/models/Task.py
from django.db import models

from .Student import Student
from .Teacher import Teacher

class Task(models.Model):
    title = models.CharField(max_length=60)
    content = models.TextField(max_length=255)
    teacher = models.ForeignKey(Teacher,on_delete=models.CASCADE,related_name='tasks')
    status = models.BooleanField(default=True)
    finished = models.ManyToManyField(Student,blank=True,related_name="finished")
    unfinished = models.ManyToManyField(Student,blank=True,related_name="unfinished")

    def __str__(self):
        return self.title

File: flutter/classroom_manager/lib/models/Task.dart
import 'Student.dart';

class Task {
  int? id;
  String? title;
  String? content;
  String? createdDate;
  List<Student>? finished;
  List<Student>? unfinished;

  Task(
      {this.id,
      this.title,
      this.content,
      this.createdDate,
      this.finished,
      this.unfinished});

  Task.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    title = json['title'];
    content = json['content'];
    createdDate = json['created_date'];
    if (json['finished'] != null) {
      finished = <Student>[];
      json['finished'].forEach((v) {
        finished!.add(Student.fromJson(v));
      });
    }
    if (json['unfinished'] != null) {
      unfinished = <Student>[];
      json['unfinished'].forEach((v) {
        unfinished!.add(Student.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['title'] = title;
    data['content'] = content;
    data['created_date'] = createdDate;
    if (finished != null) {
      data['finished'] = finished!.map((v) => v.toJson()).toList();
    }
    if (unfinished != null) {
      data['unfinished'] = unfinished!.map((v) => v.toJson()).toList();
    }
    return data;
  }
}
